bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/cloud/MiCloudConnector.java

```diff
@@ -396,22 +414,21 @@ private String loginStep1() throws InterruptedException, TimeoutException, Execu
         logger.trace("Xiaomi Login step 1 response = {}", responseStep1);
         try {
             JsonElement resp = new JsonParser().parse(parseJson(content));
-            if (resp.getAsJsonObject().has("_sign")) {
+            if (resp.isJsonObject() && resp.getAsJsonObject().has("_sign")) {
                 String sign = resp.getAsJsonObject().get("_sign").getAsString();
                 logger.trace("Xiaomi Login step 1 sign = {}", sign);
                 return sign;
             } else {
-                logger.trace("Xiaomi Login _sign missing. Maybe still has login cookie.");
+                logger.debug("Xiaomi Login _sign missing. Maybe still has login cookie.");
                 return "";
             }
-
-        } catch (JsonSyntaxException | NullPointerException e) {
+        } catch (JsonParseException | IllegalStateException | ClassCastException e) {
```

Comment:
What would throw a ClassCastException or IllegalStateException here?

Reply:
as before comment, the responses of Xiaomi cloud are many times corrupt.
maybe I'm overly caution here to catch all possible errors that can be thrown in this part of the process.
During my testing for this PR I had some JsonParseException & IllegalStateExceptions in the module, not 100% sure anymore if it was in this specific step. I may have added it to multiple steps to avoid the same.